framework/wazuh/core/batcher/client.py

```diff
@@ -0,0 +1,57 @@
+import uuid
+import asyncio
+from typing import Optional
+
+from wazuh.core.batcher.mux_demux import MuxDemuxQueue
+
+
+class BatcherClient:
+    """
+    A client for sending and receiving events via a MuxDemuxQueue.
+
+    Parameters
+    ----------
+    queue : MuxDemuxQueue
+        The MuxDemuxQueue instance used to route messages.
+    """
+    def __init__(self, queue: MuxDemuxQueue, frequency_of_wait: int = 0.1):
+        self.queue = queue
+        self.frequency_of_wait = frequency_of_wait
+
+    def send_event(self, event) -> uuid.UUID:
```

Comment:
We should use the `StatefulEvent` type from `wazuh.core.indexer.events.models`

```suggestion
    def send_event(self, event: StatefulEvent) -> uuid.UUID:
```

Reply:
Fixed in 8d89e3a86d3516f2be3d2869e9e89d12f328b223.